internal/server/authn/oidc.go

```diff
@@ -45,6 +48,18 @@ func (a *oidcAuthn) Authenticate(ctx context.Context, db *data.Transaction, requ
 		return AuthenticatedIdentity{}, fmt.Errorf("exhange code for tokens: %w", err)
 	}
 
+	if len(provider.AllowedDomains) > 0 {
+		// get the domain of the email
+		at := strings.LastIndex(email, "@") // get the last @ since the email spec allows for multiple @s
+		if at == -1 {
+			return AuthenticatedIdentity{}, fmt.Errorf("%s is an invalid email address", email)
+		}
+		domain := email[at+1:]
```

Comment:
Im going to leave this one as is since technically the email spec allows an @ in an email, even though I've never seen it. Getting the last @ means we are getting the actual domain.